src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -257,86 +258,377 @@ private struct IntPtrAndHandle
+                return MarshalType.OBJECT;
+        }
+
+        private static char? GetCallSignatureCharacterForMarshalType (MarshalType t, char? defaultValue) {
+            switch (t) {
+                case MarshalType.BOOL:
+                case MarshalType.INT:
+                case MarshalType.UINT32:
+                    return 'i';
+                case MarshalType.UINT64:
+                case MarshalType.INT64:
+                    return 'l';
+                case MarshalType.FP32:
+                    return 'f';
+                case MarshalType.FP64:
+                    return 'd';
+                case MarshalType.STRING:
+                    return 's';
+                case MarshalType.URI:
+                    return 'u';
+                case MarshalType.SAFEHANDLE:
+                    return 'h';
+                case MarshalType.ENUM:
+                    return 'j';
+                case MarshalType.ENUM64:
+                    return 'k';
+                case MarshalType.OBJECT:
+                    return 'o';
+                case MarshalType.VT:
+                    return 'a';
+                case MarshalType.POINTER:
+                    return 'm';
+                default:
+                    if (defaultValue.HasValue)
+                        return defaultValue;
+                    else
+                        throw new WasmInteropException($"Unsupported marshal type {t}");
             }
-            return new string(res);
+        }
+
+        public static string GetCallSignature(IntPtr methodHandle, object objForRuntimeType)
+        {
+            IntPtrAndHandle tmp = default(IntPtrAndHandle);
+            tmp.ptr = methodHandle;
+
+            MethodBase? mb = objForRuntimeType == null ? MethodBase.GetMethodFromHandle(tmp.handle) : MethodBase.GetMethodFromHandle(tmp.handle, Type.GetTypeHandle(objForRuntimeType));
+            if (mb == null)
+                return string.Empty;
+
+            ParameterInfo[] parms = mb.GetParameters();
+            int parmsLength = parms.Length;
+            if (parmsLength == 0)
+                return string.Empty;
+
+            var sb = new StringBuilder();
+            for (int c = 0; c < parmsLength; c++) {
+                Type t = parms[c].ParameterType;
+                var mt = GetMarshalTypeFromType(t);
+                var csc = GetCallSignatureCharacterForMarshalType(mt, null);
+                if (csc.HasValue)
+                    sb.Append(csc.Value);
+                else
+                    throw new WasmInteropException ("No signature character for marshal type " + mt);
+            }
+
+            return sb.ToString();
         }
 
         public static void SetupJSContinuation(Task task, JSObject continuationObj)
         {
+            if (task == null)
+                throw new ArgumentNullException(nameof(task));
+            if (continuationObj == null)
+                throw new ArgumentNullException(nameof(continuationObj));
```

Comment:
Use `is null` and I'd consider annotating the parameters if you expect the method to be called with null

Reply:
This is because any C# function exposed to JS can be passed null no matter what we annotate it with. What's the significance of 'is null' vs '== null?' Should I be using it everywhere? Should I annotate all of our parameters?

Reply:
I think if you are expecting null to appear as possible input I think it's worth annotating the parameters so it's clear that null handling is intentional